versioned_docs/version-v1.1/upgrade/troubleshooting.md

```diff
@@ -131,3 +131,23 @@ The Harvester controller deletes the upgrade repository VM and all files that ar
     ```
 
 4. Click the upgrade button in the Harvester dashboard to start an upgrade again.
+
+### Download upgrade relevant logs
+
+We also designed and implemented a mechanism for collecting all the upgrade-related logs along with the upgrade procedure automatically. By default, this is enabled, users can choose to opt out of such behavior though.
+
+![The "Enable Logging" checkbox on the upgrade confirmation dialog](/img/v1.1/upgrade/enable_logging.png)
+
+Users can download the log archive during an upgrade.
+
+![Download the upgrade log archive by clicking the "Download Log" button on the upgrade dialog](/img/v1.1/upgrade/download_upgradelog_dialog.png)
+
+Log entries will be collected as files for each upgrade-related Pod, even for intermediate Pods. The support bundle provides a snapshot of the current state of the cluster, including logs and resource manifests, while the upgrade log preserves any logs generated during an upgrade. By combining these two, we can look further into issues during upgrades.
+
+![The upgrade log archive contains all the logs generated by the upgrade-related Pods](/img/v1.1/upgrade/upgradelog_archive.png)
+
+After the upgrade ended, Harvester stops collecting the upgrade log automatically so it will not continue to grow and occupy the disk space. It's also worth noting that the upgrade log will be purged once users click on the "Dismiss it" button on the upgrade dialog.
```

Comment:
```suggestion
After the upgrade ended, Harvester stops collecting the upgrade logs to avoid occupying the disk space. In addition, you can click the **Dismiss it** button to purge the upgrade logs.
```